### ISSUE TITLE
Mac issue

### ISSUE THREAD

**Issue by melMass** [andrewkaufman/gafferExtensionDemo#2]
Hi Andrew,

I know you told me you don't have the time to **make this sample** working, but as I'm trying to build my OpenSubdiv extension I'm **failing** 🤕  at integrating it inside Gaffer.

I'm trying to make this sample working before putting it all together.

I want to make sure I'm on the right path.
### On mac I had to modify the CMake as following:

- Add **Gaffer Python includes**:
  ```cmake
  include_directories(${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7)
  ```
  if I don't It throw the following: 
  ```ruby
  ${GAFFER_ROOT}/include/boost/python/detail/wrap_python.hpp:50:11: fatal error:
      'pyconfig.h' file not found
  # include <pyconfig.h>
            ^~~~~~~~~~~~
  ```
- Include **boost_python IECorePython and Python** libraries:
  ```cmake
  target_link_libraries( DemoGafferExtension Gaffer GafferScene IECore IECorePython boost_python Python )
  ```
  if I don't It throw the following: 
  ```ruby
   "vtable for boost::python::objects::py_function_impl_base", referenced from:
      boost::python::objects::py_function_impl_base::py_function_impl_base() in DemoGafferExtensionModule.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function   has no definition.

  ld: symbol(s) not found for architecture x86_64
  ```

**Comment by melMass** [comment]
But on **running Gaffer** I have the following:
```python
from _DemoGafferExtension import *
ImportError: No module named _DemoGafferExtension
```

From the build dir (where _DemoGafferExtension.dylib lives) running:
```bash
gaffer env python -c 'import _DemoGafferExtension'
```

Throw the same error, so the build is'nt right.

Running:
```bash

otool -L _DemoGafferExtension returns:

```ruby
_DemoGafferExtension.dylib:
    @rpath/_DemoGafferExtension.dylib (compatibility version 0.0.0, current version 0.0.0)
    @rpath/libDemoGafferExtension.dylib (compatibility version 0.0.0, current version 0.0.0)
	lib/libGafferBindings.dylib (compatibility version 0.0.0, current version 0.0.0)
	lib/libGaffer.dylib (compatibility version 0.0.0, current version 0.0.0)
	lib/libGafferScene.dylib (compatibility version 0.0.0, current version 0.0.0)
	lib/libIECore.dylib (compatibility version 0.0.0, current version 0.0.0)
	lib/libIECorePython.dylib (compatibility version 0.0.0, current version 0.0.0)
	libboost_python.dylib (compatibility version 0.0.0, current version 0.0.0)
	/System/Library/Frameworks/Python.framework/Versions/2.7/Python (compatibility version 2.7.0, current version 2.7.10)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)

```

**Comment by andrewkaufman** [comment]
Hi, sorry I didn't see this until now. I think GitHub is not emailing me about this repo for some reason...

I'm not really aiming to support Mac here (I don't have any dev setup on a Mac), but I'm happy to take a Pull Request if you find this repro useful. I'll need to ensure it continues to work on Linux though...

As to the import issue, I think you just need to import the public module instead of the protected c++ bindings:

`gaffer env python -c 'import DemoGafferExtension; print dir(DemoGafferExtension)'`

**Comment by melMass** [comment]
Yeahhh!!

![image](https://user-images.githubusercontent.com/7041726/55796897-51475100-5acb-11e9-84fe-913771151070.png)

Thanks !!

So I found the issue... It's OSX related.
Let me diagnose it a little more and I'll pull request the new CMakeList.

But the issue was only that boost-python requires you to force `.so` vs OSX's default `dylib`.

Doing the following solved the issue:

```cmake
if(APPLE)
set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
endif(APPLE)
```

**Comment by melMass** [comment]
Let me know if #3 doesn't break on Linux

**Comment by andrewkaufman** [comment]
Fixed in #3
